OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -425,13 +428,14 @@ public ProductionItem(ProductionQueue queue, string item, int cost, PowerManager
 			OnComplete = onComplete;
 			Queue = queue;
 			this.pm = pm;
+			ai = Queue.Actor.World.Map.Rules.Actors[Item];
 		}
 
 		public void Tick(PlayerResources pr)
 		{
 			if (!Started)
 			{
-				var time = Queue.GetBuildTime(Item);
+				var time = Queue.GetBuildTime(ai, ai.TraitInfo<BuildableInfo>());
```

Comment:
Can't do this one, I think. Caching this in the constructor would mean that classic queue speedups would be applied at the moment of queuing the actor instead of when the building starts.